### ISSUE TITLE
Improve Positionable

### ISSUE THREAD

**Issue by dcampbell24** [PistonDevelopers/conrod#349]
The functions left and up will result in overlapping rectangles because the size of the rectangle being placed is not being taken into account. I think all of this positioning logic should be implemented on graphics::Rect:

```rust
impl Rect {
    fn move_above(&mut self, other: &Rect, pad: Scalar) { ... }
    fn move_below(&mut self, other: &Rect, pad: Scalar) { ... }
    fn move_left_of(&mut self, other: &Rect, pad: Scalar) { ... }
    fn move_right_of(&mut self, other: &Rect, pad: Scalar) { ... }
}

**Comment by davekong** [comment]
CC: #348

**Comment by mitchmindtree** [comment]
Hmm it used to be taken into account... I think I remember finding it changed at some point a while ago. Anyway, definitely a bug.

I just finished porting across Elm's `Element` module which has some great functionality for dealing with this, i.e. [`above`/`below`/`beside`](https://github.com/mitchmindtree/elmesque/blob/master/src/element.rs#L161), [`flow`/`layer`](https://github.com/mitchmindtree/elmesque/blob/master/src/element.rs#L294) and [various other positioning helpers](https://github.com/mitchmindtree/elmesque/blob/master/src/element.rs#L325). Still haven't gotten around to any ui implementation yet, but it's getting close.

**Comment by bvssvni** [comment]
Alternative names: `up`, `down`, `left` and `right`.

**Comment by bvssvni** [comment]
I opened https://github.com/PistonDevelopers/conrod/issues/350

**Comment by mitchmindtree** [comment]
Closed by #385, see the [`Positionable` trait](https://github.com/PistonDevelopers/conrod/blob/master/src/position.rs#L97) for all of the new options for widget and canvas layout.
